docs/content/en/docs/concepts/tracing-policy/hooks.md

```diff
@@ -158,9 +158,24 @@ spec:
 ```
 ## Uprobes
 
-{{% pageinfo %}}
-This hook point method lacks documentation, see [issue #878](https://github.com/cilium/tetragon/issues/878).
-{{% /pageinfo %}}
+## Uprobes
+
+Uprobes are similar to kprobes, but they allow you to dynamically hook into any user-space function and execute BPF code. Uprobes are also tied to the binary version of the user-space program, so they may not be portable across different versions or architectures.
+
+To use uprobes, you need to specify the path to the executable or library file, and the symbol of the function you want to probe. You can use tools like objdump, nm, or readelf to find the symbol of a function in a binary file. Alternatively, you can use the perf tool to list the available uprobes for a running process.
+
+You can define multiple uprobes in the same policy, or in different policies. You can also combine uprobes with kprobes and tracepoints to get a comprehensive view of the system behavior.
+
+Here is an example of a policy that defines an uprobe for the readline function in the bash executable, and applies it to all processes that use the bash binary:
+
+```yaml
+spec:
+  uprobes:
+  - path: "/bin/bash"
+    symbol: "readline"
+```
+
+This example shows how to use uprobes to capture the user input command line string from all bash shells. The readline function is a library function that reads a line from the standard input and returns it as a string. By hooking into this function, we can monitor the commands entered by the users.
```

Comment:
proposition: this would be nice to have an example of output from `tetra getevents -o compact` when you apply this kind of policy.

Reply:
If I see through the docs ```tetra getevents -o compact``` is used for capturing events and tracingpolicy we are defining is used to take some action based on any event right so here it might not be necessary that this policy would print the events and we should show the output if I'm not wrong . Please do correct thanks 😊

Reply:
So my idea was to get a complete example of how to use uprobe from start to finish. You gave the example of the uprobe on readline in bash.

1. Search and make sure the symbol exists in your binary (the thing you did with the `nm` command)
2. Write the TracingPolicy (the example you showed)
3. See how to see the result in action, so basically show what it would look like if you started Tetragon with that policy, wrote a few commands in a bash shell and show the output from `tetra getevents -o compact`.

Reply:
I can suggest something if it's not very clear, tell me :)

Reply:
yes I wanted to see what matchActions you would configure

Reply:
yes please . In the context of k8s

Reply:
```suggestion
This example shows how to use uprobes to hook into the readline function
running in all the bash shells.
```
Let's be succinct as we can't retrieve the args of readline as of now.